pkg/kubelet/kuberuntime/kuberuntime_manager.go

```diff
@@ -761,13 +764,22 @@ func (m *kubeGenericRuntimeManager) computePodActions(pod *v1.Pod, podStatus *ku
 						copy(changes.ContainersToUpdate[rName][1:], changes.ContainersToUpdate[rName])
 						changes.ContainersToUpdate[rName][0] = cUpdateInfo
 					}
-					return false
 				}
 				specLim := container.Resources.Limits
+				specReq := container.Resources.Requests
 				statusLim := apiContainerStatus.Resources.Limits
-				restartMem := determineContainerResize(v1.ResourceMemory, specLim.Memory().Value(), statusLim.Memory().Value())
-				restartCpu := determineContainerResize(v1.ResourceCPU, specLim.Cpu().MilliValue(), statusLim.Cpu().MilliValue())
-				if restartMem || restartCpu {
+				statusReq := apiContainerStatus.Resources.Requests
+				// Runtime container status resources, if set, takes precedence.
+				if containerStatus.Resources.Limits != nil {
+					statusLim = containerStatus.Resources.Limits
+				}
+				if containerStatus.Resources.Requests != nil {
+					statusReq = containerStatus.Resources.Requests
+				}
+				resizeMemLim, restartMemLim := determineContainerResize(v1.ResourceMemory, specLim.Memory().Value(), statusLim.Memory().Value())
```

Comment:
any reason there is not resizeMemReq and restartMemReq ?

Reply:
Currently cgroup does not support configuring memory requests.